src/e2e/java/teammates/e2e/cases/e2e/BaseE2ETestCase.java

```diff
@@ -146,4 +157,45 @@ protected AdminHomePage loginAdmin() {
         return loginAdminToPage(createUrl(Const.WebPageURIs.ADMIN_HOME_PAGE), AdminHomePage.class);
     }
 
+    /**
+     * Deletes file with fileName from the downloads folder.
+     */
+    protected void deleteDownloadsFile(String fileName) {
+        String filePath = getTestDownloadsFolder() + fileName;
+        FileHelper.deleteFile(filePath);
+    }
+
+    /**
+     * Verify file with fileName is in the downloads folder.
+     */
+    protected void verifyDownloadedFile(String expectedFileName) {
+        Path path = Paths.get(getTestDownloadsFolder() + expectedFileName);
+        assertTrue(Files.exists(path));
+    }
+
+    /**
+     * Verifies that email with subject is found in the student inbox.
+     * Student email used must be an authentic gmail account.
+     */
+    protected void verifyEmailSent(StudentAttributes student, String subject) {
+        if (TestProperties.isDevServer()) {
+            return;
+        }
+        EmailAccount email = new EmailAccount(student.getEmail());
+        try {
+            email.getUserAuthenticated();
+            int retryLimit = 5;
+            boolean actual = email.isEmailWithSubjectPresent(subject);
+            while (!actual && retryLimit > 0) {
+                retryLimit--;
+                ThreadHelper.waitFor(1000);
+                actual = email.isEmailWithSubjectPresent(subject);
+            }
+            email.markAllUnreadEmailAsRead();
```

Comment:
Is this necessary?

Reply:
It is not really necessary but is there because the checks in `EmailAccount` only looks at unread mails and so the mail is marked as read after the test to reduce the searching.